python/cudf/benchmarks/common/utils.py

```diff
@@ -0,0 +1,268 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+
+"""Common utilities for fixture creation and benchmarking."""
+
+import inspect
+import re
+import textwrap
+from collections.abc import MutableSet
+from itertools import groupby
+from numbers import Real
+
+import pytest_cases
+from config import NUM_COLS, NUM_ROWS, cudf, cupy
+
+
+def make_gather_map(len_gather_map: Real, len_column: Real, how: str):
+    """Create a gather map based on "how" you'd like to gather from input.
+    - sequence: gather the first `len_gather_map` rows, the first thread
+                collects the first element
+    - reverse:  gather the last `len_gather_map` rows, the first thread
+                collects the last element
+    - random:   create a pseudorandom gather map
+
+    `len_gather_map`, `len_column` gets rounded to integer.
+    """
+    len_gather_map = round(len_gather_map)
+    len_column = round(len_column)
+
+    rstate = cupy.random.RandomState(seed=0)
+    if how == "sequence":
+        return cudf.Series(cupy.arange(0, len_gather_map))
+    elif how == "reverse":
+        return cudf.Series(
+            cupy.arange(len_column - 1, len_column - len_gather_map - 1, -1)
+        )
+    elif how == "random":
+        return cudf.Series(rstate.randint(0, len_column, len_gather_map))
+
+
+def make_boolean_mask_column(size):
+    rstate = cupy.random.RandomState(seed=0)
+    return cudf.core.column.as_column(rstate.randint(0, 2, size).astype(bool))
+
+
+def flatten(xs):
+    for x in xs:
+        if not isinstance(x, str):
+            yield from x
+        else:
+            yield x
```

Comment:
It looks like this is valid syntax:
```python
In [5]: def flatten(xs):
   ...:     for x in xs:
   ...:         (yield x) if isinstance(x, str) else (yield from x)
```

Reply:
It is. More importantly though, `flatten` is no longer needed :upside_down_face: I changed the part of the code that was originally using it, and I can delete this altogether now.